tests/python_package_test/test_engine.py

```diff
@@ -938,6 +938,52 @@ def test_early_stopping_via_global_params(first_metric_only):
     assert "error" in gbm.best_score[valid_set_name]
 
 
+@pytest.mark.parametrize("early_stopping_round", [-10, -1, 0, None, "None"])
+def test_early_stopping_non_positive_values(early_stopping_round):
+    X, y = load_breast_cancer(return_X_y=True)
+    num_trees = 5
+    params = {
+        "num_trees": num_trees,
+        "objective": "binary",
+        "metric": "None",
+        "verbose": -1,
+        "early_stopping_round": early_stopping_round,
+        "first_metric_only": True,
+    }
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    lgb_train = lgb.Dataset(X_train, y_train)
+    lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train)
+    valid_set_name = "valid_set"
+
+    if early_stopping_round is None:
+        gbm = lgb.train(
+            params,
+            lgb_train,
+            feval=[decreasing_metric, constant_metric],
+            valid_sets=lgb_eval,
+            valid_names=valid_set_name,
+        )
+        assert "early_stopping_round" not in gbm.params
+    elif early_stopping_round == "None":
+        with pytest.raises(TypeError):
+            gbm = lgb.train(
+                params,
+                lgb_train,
+                feval=[decreasing_metric, constant_metric],
```

Comment:
```suggestion
                feval=[constant_metric],
```

Reply:
I see you did this in https://github.com/microsoft/LightGBM/pull/6406/commits/260fb9483910f7e52d15365b39448d5adef5ba8c, thanks!

In the future, I recommend using the `Add suggestion to batch` and then `Commit suggestion` buttons here on GitHub on these code suggestion comments.

Doing that automatically resolves the comment thread, so reviewers don't have to look at the diff separately to check that you applied the requested change.